### ISSUE TITLE
[Bug]: Pimcore dynamic routing (and other routing-related listeners) invoked even for system routes

### ISSUE THREAD

**Issue by dkarlovi** [pimcore/pimcore#13587]
### Expected behavior

System routes should be resolved ASAP, without invoking `Pimcore\Routing\DynamicRouteProvider::getRouteCollectionForRequest`, `\Pimcore\Model\Document\Service::getNearestDocumentByPath`, etc.

### Actual behavior

System routes like `/_fragment` used for [Symfony's ESI implementation](https://symfony.com/doc/current/http_cache/esi.html) still triggers Pimcore's routing even though it's not required in this case.

### Steps to reproduce

1. generate a fragment request which triggers a controller like so

```php
class Controller
{
    public function invokeAction(): Reponse
    {
        return new Response('AAA');
    }
}
```
2. generate a new fragment URL (for example, using `render_esi` Twig function), it will looks something like
```
/_fragment?_hash=ABC&_controller%3DController%253A%253AinvokeAction
```
3. access the URL and notice it's behaving as if this route is dynamic: loading routes from the database, checking against possible documents named like that (which shouldn't even be allowed since it would break this feature).

Optimizing this endpoint specifically is crucial to be able to use Pimcore efficiently with ESI since there could be many ESI requests required to fulfil the response in certain scenarios, every ms saved here makes a difference.

**Comment by dvesh3** [comment]
While working on the fix, I found out that it is possible to create a pimcore document with path `/_fragment` or even a static route so in these cases the dynamic routing is required otherwise request will not be resolved. This should only be skipped when using the `ESI fragment`, which can also have a different path like explained here https://symfony.com/doc/current/http_cache/esi.html#using-esi-in-symfony

**Comment by dkarlovi** [comment]
@dvesh3 if you have real Symfony routes defined, they're defined at build time and chances are they **must** work. Being able to override Symfony routes with a dynamic document seems like ripe for possible user-generated errors.

For example, imagine having a user creating a document in place of Pimcore's API and then being unable to delete it because now the API is broken / inaccessible.

**Comment by dvesh3** [comment]
Fixed by #13880
